python-avd/pyavd/_eos_designs/structured_config/underlay/prefix_lists.py

```diff
@@ -21,66 +22,55 @@ class PrefixListsMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> list | None:
-        """Return structured config for prefix_lists."""
-        if self.shared_utils.underlay_bgp is not True and not self.shared_utils.is_wan_router:
-            return None
-
-        if self.shared_utils.overlay_routing_protocol == "none" and not self.shared_utils.is_wan_router:
-            return None
+    @structured_config_contributor
+    def prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> None:
+        """Set the structured config for prefix_lists."""
+        if not self.shared_utils.is_wan_router and (not self.shared_utils.underlay_bgp or self.shared_utils.overlay_routing_protocol == "none"):
+            return
 
         if not self.inputs.underlay_filter_redistribute_connected:
-            return None
+            return
 
         # IPv4 - PL-LOOPBACKS-EVPN-OVERLAY
-        sequence_numbers = [
-            {"sequence": index * 10, "action": f"permit {network} eq 32"}
-            for index, network in enumerate(collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.loopback_ipv4_pool)), start=1)
-        ]
+        sequence_numbers = EosCliConfigGen.PrefixListsItem.SequenceNumbers()
+        for index, network in enumerate(collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.loopback_ipv4_pool)), start=1):
+            sequence_numbers.append_new(sequence=index * 10, action=f"permit {network} eq 32")
 
         if self.shared_utils.overlay_vtep and self.shared_utils.vtep_loopback.lower() != "loopback0" and not self.shared_utils.is_wan_router:
-            sequence_numbers.extend(
-                {"sequence": index * 10, "action": f"permit {network} eq 32"}
-                for index, network in enumerate(
-                    collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.vtep_loopback_ipv4_pool)), start=len(sequence_numbers) + 1
-                )
-            )
+            for index, network in enumerate(
+                collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.vtep_loopback_ipv4_pool)), start=len(sequence_numbers) + 1
+            ):
+                sequence_numbers.append_new(sequence=index * 10, action=f"permit {network} eq 32")
 
         if self.inputs.vtep_vvtep_ip is not None and self.shared_utils.network_services_l3 is True and not self.shared_utils.is_wan_router:
             sequence_number = (len(sequence_numbers) + 1) * 10
-            sequence_numbers.append({"sequence": sequence_number, "action": f"permit {self.inputs.vtep_vvtep_ip}"})
-
-        prefix_lists = [{"name": "PL-LOOPBACKS-EVPN-OVERLAY", "sequence_numbers": sequence_numbers}]
+            sequence_numbers.append_new(sequence=sequence_number, action=f"permit {self.inputs.vtep_vvtep_ip}")
```

Comment:
no need to redefine sequence_number you can just put the expression inside the append_new